.github/workflows/ai_benchmark_nsql.yml

```diff
@@ -0,0 +1,154 @@
+          chmod +x testoperator
+          
+      - name: Save spice artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: testoperator
+          path: testoperator
+
+  build-spiced:
+    name: Build spiced
+    runs-on: 'spiceai-macos'
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: ./.github/actions/setup-rust
+        with:
+          os: darwin
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v3
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build spiced
+        if: matrix.target.target_os != 'darwin'
+        run: |
+          make -C bin/spiced SPICED_NON_DEFAULT_FEATURES="models"
+
+      - name: make spiced executable
+        if: matrix.target.target_os != 'windows'
+        run: |
+          mv target/release/spiced spiced
+          chmod +x spiced
+
+      - name: Save spice artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: spiced
+          path: spiced
+
+  setup-matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.setup-matrix.outputs.result }}
+
+    steps:
+      - name: Set up matrix
+        uses: actions/github-script@v7
+        id: setup-matrix
+        with:
+          script: |
+            const matrix = [
+              {
+                name: "gpt-4o-mini",
+                spicepod: "./test/nsql_bench/spicepod.yaml",
+              },
+              {
+                name: "llama3-3B-Instruct",
+                spicepod: "./test/nsql_bench/spicepod_hf.yaml",
+              },
+            ];
+
+            const model_option = context.payload.inputs.model_option;
+            let filtered = matrix;
+
+            if (model_option !== 'all') {
+              filtered = filtered.filter(m => m.name === model_option);
+            }
+
+            return filtered;
+
+  run-testoperator:
+    name: tpch_nsql benchmark
+    needs:
+      - build-testoperator
+      - build-spiced
+      - setup-matrix
+    runs-on: 'spiceai-macos'
+    strategy:
+      fail-fast: false
+      matrix:
+        target: ${{ fromJson(needs.setup-matrix.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download testoperator
+        uses: actions/download-artifact@v4
+        with:
+          name: testoperator
+  
+      - name: Download spiced
+        uses: actions/download-artifact@v4
+        with:
+          name: spiced
+
+      - name: Mark binaries executable
+        run: |
+          chmod +x ./testoperator
+          chmod +x ./spiced
+
+      - name: Run tpch_nsql evals
+        env:
+          SPICE_OPENAI_API_KEY: ${{ secrets.SPICE_SECRET_OPENAI_API_KEY }}
+          SPICE_HF_TOKEN: ${{ secrets.SPICE_SECRET_HF_TOKEN }}
+        run: |
+          ./testoperator run evals \
+            -s $GITHUB_WORKSPACE/spiced \
+            --eval tpch_nsql \
```

Comment:
I think this GH action could be easily broadened to input both a model option, and an eval task (instead of hardcoded `tpch_nsql`).

Reply:
Simplified so that model and eval are optional and the testoperator will default to spicepod definition (use first found model and eval if not specified). GH workflow has been updated so that more eval configurations could be added (combination of spicepod + data folder)